docs/src/introduction_to_catalyst/advanced_intro_to_catalyst.md

```diff
@@ -0,0 +1,356 @@
+
+This advanced introduction is not required for understanding the remaining documentation, nor for using any of Catalyst's features. It is not expected that new users will read it. Instead, it aims to provide additional context and understanding for more experienced users. If you have just started exploring how to use Catalyst, reading this section is not needed. However, if you are using (or plan to use) Catalyst extensively, reading this page will be useful.
+
+## [How Catalyst is constructed](@id advanced_intro_to_catalyst_background)
+Catalyst is built on top of the domain-agnostic modelling package [ModelingToolkit.jl](https://github.com/SciML/ModelingToolkit.jl). ModelingToolkit is in turn built on the computer algebraic system [Symbolics.jl](https://github.com/JuliaSymbolics/Symbolics.jl). Here we will introduce these packages, and how the feed into Catalyst.
+
+### [The Symbolics.jl computer algebraic system](@id advanced_intro_to_catalyst_background_symbolics)
+Symbolics is a Julia package for representing, and doing computations on, symbolic expressions. It allow us to declare symbolic variables that can be used to construct equations. Primarily this is done through the `@variables` macro:
+```@example advanced_intro_to_catalyst_symbolics
+using Symbolics
+@variables x y
+nothing # hide
+```
+Here, both `x` and `y` becomes Julia variables in the current scope, e.g. we can do
+```@example advanced_intro_to_catalyst_symbolics
+x
+```
+to confirm that `x` is a Julia variable we can work with. Next, we can use these variables to create algebraic equations:
+```@example advanced_intro_to_catalyst_symbolics
+eq = x^2 - y^2
+```
+Symbolics's primary feature is that it enables algebraic manipulation on algebraic expressions. We can use its `simplify` function to get a simpler (but equivalent) expression. E.g. it can correctly apply the [difference of two squares](https://en.wikipedia.org/wiki/Difference_of_two_squares) rule to our previous equation:
+```@example advanced_intro_to_catalyst_symbolics
+simplify(eq / (x + y))
+```
+
+!!! warning
+    `@variables` (and similar [macros introduced later on](@ref ref), like `@parameters` and `@species`) create new variables in the current scope. E.g. if you have previously defined `x = 5` and then run `@variables x`, this will overwrite the previous value of `x`.
+
+Symbolics is used internally within Catalyst to represent all models. One advantage of this is that complicated algebraic expressions can be automatically simplified. E.g. if we create a model with a (trivially) unnecessarily complicated rate and check what it is, we note that it has been automatically simplified
+```@example advanced_intro_to_catalyst_symbolics
+using Catalyst
+rn = @reaction_network begin
+    (k1*k2)/k2, X --> 0
+end
+reactions(rn)[1].rate
+```
```

Comment:
Here I had more on symbolic simplifications performed, but looking at examples, it seems that these are never really done (even with `structural_simplify`).